### DISCUSSION TITLE
return 501 instead of 404

### DISCUSSION THREAD

**Discussion by ahakem** [vercel/next.js#16157] [Help] [answered]
Hello,
for some reason, I need to return 501 instead of 404 for all not existing pages so another app managed via Nginx will take care of this 501 to display it or return 404 if not exist.

IS any way to do that?

**Comment by ovidyou** [comment, marked as answer]
@ahakem 

Create an `_error.js` page with following content:

```
// ./page/_error.js

function Error({ statusCode }) {
  return (
    <p>
      {statusCode
        ? `An error ${statusCode} occurred on server`
        : "An error occurred on client"}
    </p>
  );
}

export async function getServerSideProps({ res, err }) {
  let statusCode = res ? res.statusCode : err ? err.statusCode : 404;

  if (statusCode === 404) {
    statusCode = 501;
    res.statusCode = 501;
    res.end();
  }

  return { statusCode };
}

export default Error;
```

If you already have a `404.js` page, then delete it, otherwise it will not work.

**Reply by ahakem** [reply]
Thank you, I'll try that
